framework/contracts/native/account-factory/src/commands.rs

```diff
@@ -107,123 +105,124 @@ pub fn execute_create_account(
             ))
         })?;
     }
-    // save module for after-init check
-    CONTEXT.save(
-        deps.storage,
-        &Context {
-            account_id: account_id.clone(),
-            account_proxy_address: None,
-            manager_module: None,
-            proxy_module: Some(module.clone()),
-
-            additional_config: AdditionalContextConfig {
-                namespace,
-                base_asset,
-                name,
-                description,
-                link,
-                owner: governance.into(),
-            },
-            install_modules,
-            funds_for_install,
-        },
-    )?;
 
-    if let ModuleReference::AccountBase(proxy_code_id) = module.reference {
-        Ok(AccountFactoryResponse::new(
-            "create_account",
-            vec![
-                ("account_sequence", &account_id.seq().to_string()),
-                ("trace", &account_id.trace().to_string()),
-            ],
-        )
-        // Create proxy
-        .add_submessage(SubMsg {
-            id: CREATE_ACCOUNT_PROXY_MSG_ID,
-            gas_limit: None,
-            msg: WasmMsg::Instantiate {
-                code_id: proxy_code_id,
-                funds: funds_to_proxy.into_vec(),
-                // Currently set admin to self, update later when we know the contract's address.
-                admin: Some(env.contract.address.to_string()),
-                label: format!("Proxy of Account: {}", account_id),
-                msg: to_json_binary(&ProxyInstantiateMsg {
-                    account_id,
-                    ans_host_address: config.ans_host_contract.to_string(),
-                })?,
-            }
-            .into(),
-            reply_on: ReplyOn::Success,
-        }))
+    let salt = generate_salt(env.block.height, &account_id);
```

Comment:
For this application there shouldn't be a need to add the block height to the salt because for every account you can only create once and there's no re-install possible. Removing the height would make it easy to figure out what the proxy address for any account (also interchain accounts) will be. 

I.e. we can send funds to an address that we know will be assigned to the proxy of an account on chain B while the original account is on chain A. 

Also in the `generate_salt` function we take a part of the account's trace to create the salt. I think we should hash the trace and use that instead. That allows the trace to be of any length.

Reply:
Or we can just pass zero as block height here, wdyt?

Reply:
sha256 is pretty quick and easy

Reply:
> Or we can just pass zero as block height here, wdyt?

I'd prefer having different options (fns). I feel like we might want more options in the future.

Reply:
Using hash now and added different fn for account address salt